woocommerce/wc-admin
# Add Single Date Picker

Working towards #1027 
Depends on https://github.com/WordPress/gutenberg/pull/12855

![screen shot 2018-12-14 at 1 37 53 pm](https://user-images.githubusercontent.com/1922453/49976109-81f7fd80-ffa5-11e8-943d-54babbaa1094.png)

![screen shot 2018-12-14 at 1 38 37 pm](https://user-images.githubusercontent.com/1922453/49976139-9e943580-ffa5-11e8-8a54-a079e7c6fb83.png)

## Test

1. Checkout `psealock:add/focusOnMount-to-dropdown` [branch](https://github.com/psealock/gutenberg/tree/add/focusOnMount-to-dropdown) of Gutenberg or just make the two line change in https://github.com/WordPress/gutenberg/pull/12855.
1. View Calendar in DevDocs `/wp-admin/admin.php?page=wc-admin#/devdocs/calendar`
2. Play around with the single date picker
    * Mobile sizes
    * Keyboard interactions
    * VoiceOver
3. If you see a propType warning for `block`, the fix for that is here https://github.com/WordPress/gutenberg/pull/12933

## Key Decisions

* **Make the react-dates datepicker unreachable by keyboard**. There is no way to simultaneously keep focus on the input so keyboard users can type inputs and allow focus on the dropdown for selection via keyboard. One solution is to keep focus on the input, then allow focus on the dropdown via tabbing. I don't like this because it misrepresents the sequential flow of a form and causes confusion, for me at least. React-dates [implementation](https://airbnb.io/react-dates/?selectedKind=SingleDatePicker%20%28SDP%29&selectedStory=default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel) chooses similar behaviour.
* **Don't allow full screen expansion of dropdown on mobile screens**. Otherwise, it forces focus away from the input field, effectively removing the ability to use the keyboard.
